src/imageops/fast_blur.rs

```diff
@@ -0,0 +1,144 @@
+            &samples,
+            width as usize,
+            height as usize,
+            *radius,
+            P::CHANNEL_COUNT as usize,
+        );
+        samples = my_fast_horizontal_blur::<P::Subpixel>(
+            &horizontally_blurred_transposed,
+            height as usize,
+            width as usize,
+            *radius,
+            P::CHANNEL_COUNT as usize,
+        );
+    }
+    ImageBuffer::from_raw(width, height, samples).unwrap()
+}
+
+fn boxes_for_gauss(sigma: f32, n: usize) -> Vec<usize> {
+    let w_ideal = f32::sqrt((12.0 * sigma * sigma / (n as f32)) + 1.0);
+    let mut w_l = w_ideal.floor();
+    if w_l % 2.0 == 0.0 {
+        w_l -= 1.0
+    };
+    let w_u = w_l + 2.0;
+
+    let m_ideal = (12.0 * sigma * sigma
+        - (n as f32) * (w_l) * (w_l)
+        - 4.0 * (n as f32) * (w_l)
+        - 3.0 * (n as f32))
+        / (-4.0 * (w_l) - 4.0);
+    let m = f32::round(m_ideal) as usize;
+
+    let mut box_sizes: Vec<usize> = vec![];
+    for i in 0..n {
+        box_sizes.push(if i < m { w_l as usize } else { w_u as usize });
+    }
+    box_sizes
+}
+
+fn channel_idx(channel: usize, idx: usize, channel_num: usize) -> usize {
+    channel_num * idx + channel
+}
+
+fn my_fast_horizontal_blur<P: Primitive>(
+    samples: &[P],
+    width: usize,
+    height: usize,
+    r: usize,
+    channel_num: usize,
+) -> Vec<P> {
+    let channel_size = width * height;
+
+    let mut out_samples: Vec<P> = vec![P::from(0).unwrap(); channel_size * channel_num];
+    let mut vals = vec![0.0; channel_num];
+
+    let min_value = P::DEFAULT_MIN_VALUE.to_f32().unwrap();
+    let max_value = P::DEFAULT_MAX_VALUE.to_f32().unwrap();
+
+    for i in 0..height {
+        for (channel, value) in vals.iter_mut().enumerate().take(channel_num) {
+            *value = ((-(r as isize))..(r + 1) as isize)
+                .map(|x| {
+                    extended_f(samples, width, height, x, i as isize, channel, channel_num)
+                        .to_f32()
+                        .unwrap_or(0.0)
+                })
+                .sum()
+        }
+
+        for j in 0..width {
+            for channel in 0..channel_num {
+                let val = vals[channel] / (2.0 * r as f32 + 1.0);
+                let val = if val < min_value {
+                    min_value
+                } else if val > max_value {
+                    max_value
+                } else {
+                    val
+                };
+                let val = P::from(val).unwrap();
+
+                out_samples[channel_idx(channel, i + j * height, channel_num)] = val;
+                vals[channel] = vals[channel]
+                    - extended_f(
+                        samples,
+                        width,
+                        height,
+                        j as isize - r as isize,
+                        i as isize,
+                        channel,
+                        channel_num,
+                    )
```

Comment:
nit: more readable with intermediate variable names for this and the mirrored invocation.

Reply:
fixed